test-coverage-enforcer/src/main/java/org/wso2/testcoverageenforcer/Maven/Jacoco/JacocoCoverage.java

```diff
@@ -0,0 +1,613 @@
+                        nodeAnalysisReport.put(Constants.Jacoco.JACOCO_GOAL_COVERAGE_RULE_INVOKE, true);
+                        break;
+                }
+            }
+            // Append each template Node for missing jacoco nodes
+            if (!(nodeAnalysisReport.get(Constants.Jacoco.JACOCO_GOAL_AGENT_INVOKE))) {
+                Node jacocoPrepareAgentExecutionNodeTemplate = pom.importNode(TemplateReader.extractTemplate(Constants.Jacoco.JACOCO_PREPARE_AGENT_TEMPLATE), true);
+                executions.appendChild(jacocoPrepareAgentExecutionNodeTemplate);
+                log.debug("Jacoco prepare-agent execution template added in under " + pluginsParent + " for " + pom.getDocumentURI());
+            }
+            if (!(nodeAnalysisReport.get(Constants.Jacoco.JACOCO_GOAL_REPORT))) {
+                Node jacocoReportExecutionNodeTemplate = pom.importNode(TemplateReader.extractTemplate(Constants.Jacoco.JACOCO_REPORT_TEMPLATE), true);
+                Element jacocoReportExecutionNodeElement = (Element) jacocoReportExecutionNodeTemplate;
+                jacocoReportExecutionNodeElement.getElementsByTagName(Constants.Jacoco.JACOCO_TAG_REPORT_READ).item(0)
+                        .setTextContent(jacocoReportFilePath);
+                executions.appendChild(jacocoReportExecutionNodeTemplate);
+                log.debug("Jacoco report execution template added in under " + pluginsParent + " for " + pom.getDocumentURI());
+            }
+            if (!(nodeAnalysisReport.get(Constants.Jacoco.JACOCO_GOAL_COVERAGE_RULE_INVOKE))) {
+                Node jacocoCheckExecutionNodeTemplate = pom.importNode(TemplateReader.extractTemplate(Constants.Jacoco.JACOCO_CHECK_TEMPLATE), true);
+                Element jacocoCheckExecutionElementTemplate = (Element) jacocoCheckExecutionNodeTemplate;
+                jacocoCheckExecutionElementTemplate.getElementsByTagName(Constants.Jacoco.JACOCO_TAG_COVERAGE_PER_ELEMENT).item(0)
+                        .setTextContent(coveragePerElement);
+                jacocoCheckExecutionElementTemplate.getElementsByTagName(Constants.Jacoco.JACOCO_TAG_COVERAGE_CHECK_VALUE).item(0)
+                        .setTextContent(coverageThreshold);
+                jacocoCheckExecutionElementTemplate.getElementsByTagName(Constants.Jacoco.JACOCO_TAG_REPORT_READ).item(0)
+                        .setTextContent(jacocoReportFilePath);
+                executions.appendChild(jacocoCheckExecutionNodeTemplate);
+                log.debug("Jacoco check execution template added in under " + pluginsParent + " for " + pom.getDocumentURI());
+            }
+            if (
+                    nodeAnalysisReport.get(Constants.Jacoco.JACOCO_GOAL_AGENT_INVOKE) &&
+                            nodeAnalysisReport.get(Constants.Jacoco.JACOCO_GOAL_REPORT) &&
+                            nodeAnalysisReport.get(Constants.Jacoco.JACOCO_GOAL_COVERAGE_RULE_INVOKE)) {
+                log.debug("Jacoco Plugin already available");
```

Comment:
Add a condition to check whether the debug logs are enabled or not? and check this through the PR

Reply:
Fixed in 2bc0618a39bb056608ff116edf53bc2b3f4e8099.